Tests/kaas/plugin/interface.py

```diff
@@ -0,0 +1,80 @@
+from abc import ABC, abstractmethod
+from typing import final
+import os
+import shutil
+import logging
+
+logger = logging.getLogger("interface")
+
+
+class KubernetesClusterPlugin(ABC):
+    """
+    An abstract base class for custom Kubernetes cluster provider plugins.
+    It represents an interface class from which the api provider-specific
+    plugins must be derived as child classes
+    """
+
+    kubeconfig = None
+    cluster_name = None
+    k8s_api_client = None
+    working_directory = None
+
+    @final
+    def __init__(self, config=None):
+        logger.info(f"Inital provider plug in of type:{type(self)}")
+        self.config = config
+        logger.debug(self.config)
+        self.working_directory = os.getcwd()
+        logger.debug(f"Working from {self.working_directory}")
+
+    @abstractmethod
+    def _create_cluster(self, cluster_name) -> (str, int):
+        """
+        Create a Kubernetes cluster to test aggainst.
+        :param: cluster_name:
+        :return: kubeconfig: kubeconfig of the cluster used for testing
+        """
+        pass
+
+    @abstractmethod
+    def _delete_cluster(self, cluster_name) -> (str, int):
+        """
+        Delete the Kubernetes cluster.
+        :param: cluster_name:
+        :return: None
+        """
+        pass
+
+    @final
+    def create(self, name="scs-cluster", version=None, kubeconfig_filepath=None):
```

Comment:
This would also include removing the content of the methods. I wanted to use this function to give an outline of what a create function should do. (return a kubeconfig file)
Would you also recommend removing this?

Reply:
Yes, of course. A good docstring and our subclasses will suffice.

Reply:
I have adapted it accordingly.